senior_teacher.rb

```diff
@@ -1,22 +1,19 @@
-class SeniorTeacher
-  attr_reader :age, :salary, :phase, :performance_rating, :target_raise
-  attr_accessor :name
+require_relative 'human'
+
+class SeniorTeacher < Teacher
+  include HighFiveable
+  include Payable
```

Comment:
In object oriented programming, we tend to prefer using inheritance over modules. This way will certainly work, but we generally only use modules for things that aren't central to the object they live in, or which don't make sense in an existing superclass (and it doesn't make sense to add in a middle layer superclass). In this case, this behavior is very central to what it means to be a teacher at dbc, so they should be part of the superclass for apprentice and senior teacher, not a separate module.